steps/terraform.yaml

```diff
@@ -0,0 +1,149 @@
+
+  - task: TerraformCLI@0
+    displayName: Terraform plan ${{ parameters.stack }}
+    condition: and(succeeded(), in('${{ parameters.overrideAction }}', 'plan', 'apply'))
+    inputs:
+      command: plan
+      terraformVersion: ${{ parameters.tfversion }}
+      workingDirectory: '$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/components/${{ parameters.stack }}'
+      environmentServiceName: ${{ parameters.serviceConnection }}
+      commandOptions: >
+        -out tfplan-${{ parameters.environment }}-${{ parameters.stack }}
+        -var env=${{ parameters.environment }}
+        -var builtFrom=$(Build.Repository.Name)
+        -var product=${{ parameters.product }} ${{ parameters.planCommandOptions }}
+        -var-file $(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars
+        -lock=false
+
+  - task: TerraformCLI@0
+    displayName: Terraform apply ${{ parameters.stack }}
+    condition: |
+      or(
+      and(succeeded(), eq(variables['isMain'], true), eq('${{ parameters.overrideAction }}', 'apply')),
+      and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true))
+      )
+    inputs:
+      command: apply
+      terraformVersion: ${{ parameters.tfversion }}
+      workingDirectory: '$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/components/${{ parameters.stack }}'
+      environmentServiceName: ${{ parameters.serviceConnection }}
+      commandOptions: '--auto-approve tfplan-${{ parameters.environment }}-${{ parameters.stack }} ${{ parameters.applyCommandOptions }}'
+
+  - ${{ if eq(parameters.overrideAction, 'destroy') }}:
+      - task: TerraformCLI@0
+        displayName: Terraform destroy ${{ parameters.stack }}
+        condition: and(succeeded(), eq(variables['isMain'], true), eq('${{ parameters.overrideAction }}', 'destroy'))
+        inputs:
+          command: destroy
+          terraformVersion: ${{ parameters.tfversion }}
+          workingDirectory: '$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/components/${{ parameters.stack }}'
+          environmentServiceName: ${{ parameters.serviceConnection }}
+          commandOptions: >
+            -var environment=${{ parameters.environment }}
+            -var builtFrom=$(Build.Repository.Name)
+            -var product=${{ parameters.product }} ${{ parameters.destroyCommandOptions }}
+            -var-file $(System.DefaultWorkingDirectory)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars
+
+  - task: Bash@3
+    displayName: "Remove local tfstate"
+    inputs:
+      targetType: 'inline'
+      workingDirectory: '$(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/components/${{ parameters.stack }}'
+      script: |
+        rm -f terraform.tfstate* && rm -rf .terraform/
+
+  - task: AzureCLI@2
+    displayName: 'Unlock TF state if required'
+    condition: always()
```

Comment:
Should we add a condition here similar to when Terraform apply is only ran? 
- Tf plan won't run due to -lock=false

Reply:
lock=false removed from plan as no longer required.